lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -619,6 +620,72 @@ class WebMapServiceCatalogItem
     }
   );
 
+  fetchPalettes(): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+      if (!this.isThredds) {
+        reject("Not a THREDDS server");
+      }
+      const urlRegex = /(https?:\/\/[^\s]+)/g;
+      const selectedStyle = this.availableStyles[0].styles.filter(
+        (style) => style.name === this.styles
+      );
+      const abstract = selectedStyle[0]?.abstract;
+      const urls = abstract?.match(urlRegex);
+      const extractedUrl = urls?.[0];
+
+      if (!extractedUrl) {
+        reject("No URL found in abstract");
```

Comment:
Unless we also return here the fetch below will still execute and make a request to `<ur>/undefined`

Reply:
Updated to resolve, thank Lawrence